src/shogun/multiclass/KNN.cpp

```diff
@@ -249,49 +315,56 @@ SGMatrix<int32_t> KNN::classify_for_multiple_k()
 
 	io::info("{} test examples", num_lab);
 
-	init_solver(m_knn_solver);
+	//init_solver(m_knn_solver);
 
 	SGVector<int32_t> output = solver->classify_objects_k(distance, num_lab, train_lab, classes);
 
 
-
+	std::cout<<"exiting KNN::classify_for_multiple_k()\n";
 	return SGMatrix<int32_t>(output,num_lab,m_k);
 }
 
 void KNN::init_distance(std::shared_ptr<Features> data)
 {
+	std::cout<<"entered KNN::init_distance(std::shared_ptr<Features> data)\n";
 	require(distance, "Distance not set.");
 	auto lhs=distance->get_lhs();
 	if (!lhs || !lhs->get_num_vectors())
 	{
 		error("No vectors on left hand side");
 	}
 	distance->init(lhs, std::move(data));
+	std::cout<<"exiting KNN::init_distance(std::shared_ptr<Features> data)\n";
 }
 
 bool KNN::load(FILE* srcfile)
 {
+	std::cout<<"entered KNN::load(FILE* srcfile)\n";
 	SG_SET_LOCALE_C;
 	SG_RESET_LOCALE;
+	std::cout<<"exiting KNN::load(FILE* srcfile)\n";
 	return false;
 }
 
 bool KNN::save(FILE* dstfile)
 {
+	std::cout<<"entered KNN::save(FILE* dstfile)\n";
 	SG_SET_LOCALE_C;
 	SG_RESET_LOCALE;
+	std::cout<<"exiting KNN::save(FILE* dstfile)\n";
 	return false;
 }
 
 void KNN::init_solver(KNN_SOLVER knn_solver)
 {
+	std::cout<<"entered KNN::init_solver(KNN_SOLVER knn_solver)\n";
 	switch (knn_solver)
 	{
 	case KNN_BRUTE:
 	{
-		SGMatrix<index_t> NN = nearest_neighbors();
-		solver = std::make_shared<BruteKNNSolver>(m_k, m_q, m_num_classes, m_min_label, m_train_labels, NN);
-
+		//SGMatrix<index_t> NN = nearest_neighbors();
+		solver = std::make_shared<BruteKNNSolver>(m_k, m_q, m_num_classes, m_min_label, m_train_labels);
```

Comment:
I changed the code for the `BruteKNNSolver` class to accommodate the new design, however, for some reason the program control isn't entering the` BruteKNNSolver()` constructor to create the `solver` instance. Can someone please help me out?

Please excuse the unnecessary print statements and comments, they are just for debugging and will be removed when merging.

Reply:
use gdb, or `SG_DEBUG` that is way quicker than recompiling and printing